src/xdocs/config_coding.xml

```diff
@@ -5849,6 +5849,133 @@ myList.stream()
+        </p>
+      </subsection>
+      <subsection name="Notes" id="UnnecessarySemicolonAfterTypeDeclaration_Notes">
+        <p>
+          This check is not applicable to nested type declarations,
+          <a
+          href="https://checkstyle.org/config_coding.html#UnnecessarySemicolonAfterTypeMemberDeclaration">
+          UnnecessarySemicolonAfterTypeMemberDeclaration</a> is responsible for it.
+        </p>
+      </subsection>
+      <subsection name="Properties" id="UnnecessarySemicolonAfterTypeDeclaration_Properties">
+        <div class="wrapper">
+          <table>
+            <tr>
+              <th>name</th>
+              <th>description</th>
+              <th>type</th>
+              <th>default value</th>
+              <th>since</th>
+            </tr>
+            <tr>
+              <td>tokens</td>
+              <td>tokens to check</td>
+              <td>
+                subset of tokens
+                <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CLASS_DEF">
+                 CLASS_DEF</a>,
+                <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INTERFACE_DEF">
+                INTERFACE_DEF</a>,
+                <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF">
+                ENUM_DEF</a>,
+                <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_DEF">
+                ANNOTATION_DEF</a>.
+              </td>
+              <td>
+                <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CLASS_DEF">
+                    CLASS_DEF</a>,
+                <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INTERFACE_DEF">
+                    INTERFACE_DEF</a>,
+                <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF">
+                    ENUM_DEF</a>,
+                <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_DEF">
+                    ANNOTATION_DEF</a>.
+              </td>
+              <td>8.31</td>
+            </tr>
+          </table>
+        </div>
+      </subsection>
+
+      <subsection name="Examples" id="UnnecessarySemicolonAfterTypeDeclaration_Examples">
+        <p>
+          To configure the check:
+        </p>
+        <source>
+&lt;module name=&quot;UnnecessarySemicolonAfterTypeDeclaration&quot;/&gt;
+        </source>
+        <p>
+          Example:
+        </p>
+        <source>
+ class A {
+
+     class Nested {
+
+     }; // OK, nested type declarations are ignored
+
+ }; // violation
+
+ interface B {
+
+ }; // violation
+
+ enum C {
+
+ }; // violation
+
+ {@literal @}interface D {
+
+ }; // violation
+        </source>
+      </subsection>
```

Comment:
Here may be an example configuration with the "tokens" property, but this will be redundant. I don't know a reason to restrict this check only for classes but not interfaces.

Reply:
I don't quite understand the point.
About only classes - report shows violations on interfaces and enums as well.
About examples of other tokens config - I think it is not necessary for such simple check.

Reply:
@pbludov Please confirm and mark resolved when it is done.

I assume he is referring to my item https://github.com/checkstyle/checkstyle/pull/7935#discussion_r396093038 .